### PR TITLE
Refactored all Constructors to Classes & Fixed Linting Errors

### LINKED ISSUE #111: Implement Classes in place of constructors
#### Nature of issue?
- [x] Existing feature enhancement

#### Feature enhancement details:
I want to replace the ES5 constructor functions with classes. I gave this a shot a few months ago but once #107 is fixed I can really make this happen. This will increase readability and just keep the code base more modern. Please assign this to me and I will work it as soon as #107 is complete.

### PR DESCRIPTION
**Pull request by spiray**
I went through all the entities and refactored the constructor functions to ES6 classes. This PR fixes #111. I rebuilt the project and tested the examples and it seems to be working well. @MathuraMG @lm-n can you review the code and run the examples to make sure that no features broke? Can you also look at the `isMember()` method in the `baseEntity` class and let me know whether I should delete it?

### CODE REVIEW

**Comment by MathuraMG** [comment]
@spiray  - will have a look today!

File: src/entities/fillEntity.js [hunk 1]
```diff
@@ -1,17 +1,16 @@
...
+  constructor(Interceptor, shapeObject, fillArgs, canvasX, canvasY) {
+    this.fillArgs = [...fillArgs];
+    this.populate(Interceptor);
+  }
+  populate(Interceptor) {
+    if (this.fillArgs[0].name === `p5.Color`) {
+      this.fillArgs = this.backgroundArgs[0].levels;
     }
-    Interceptor.currentColor = Interceptor.getColorName(fillArgs).color + Interceptor.getColorName(fillArgs).rgb;
+    Interceptor.bgColor = Interceptor.getColorName(this.fillArgs).color + Interceptor.getColorName(this.fillArgs).rgb;
```
Comment (MathuraMG) [inline src/entities/fillEntity.js]: Could you verify if this should be `Interceptor.currentColor` ?

File: src/entities/fillEntity.js [hunk 2]
```diff
@@ -1,17 +1,16 @@
...
-    if (fillArgs[0].name === `p5.Color`) {
-      fillArgs = fillArgs[0].levels;
+class FillEntity {
+  constructor(Interceptor, shapeObject, fillArgs, canvasX, canvasY) {
+    this.fillArgs = [...fillArgs];
+    this.populate(Interceptor);
+  }
+  populate(Interceptor) {
+    if (this.fillArgs[0].name === `p5.Color`) {
+      this.fillArgs = this.backgroundArgs[0].levels;
```
Comment (MathuraMG) [inline src/entities/fillEntity.js]: Could you verify if this should be `fillArgs` ?

Comment (spiray) [reply src/entities/fillEntity.js]: Thanks for finding that. It has been fixed in the next 2 commits.

File: src/entities/fillEntity.js (hunk 1 above)
Comment (spiray) [reply src/entities/fillEntity.js]: This has been fixed as well. Thanks

**Comment by spiray** [comment]
The latest 2 commits fix #125. @MathuraMG Can you review the last lint [error](https://github.com/processing/p5.accessibility/blob/0acf72edc9a670e212e3ddc03dfcb1564c636474/src/gridInterceptor/interceptorFunctions.js#L115)? It seems like it may have been there for debugging. Thanks!

**Comment by MathuraMG** [comment]
@spiray - you're right, that can be removed

**Comment by spiray** [comment]
@MathuraMG I fixed that error. If everything else looks good this can be merged

**Comment by MathuraMG** [comment]
@spiray - sorry I meant, we can remove the truthy loop itself.. btw are the examples working for you on this branch?

**Comment by spiray** [comment]
I should remove that whole `if` block? They aren't working now... :(

**Comment by MathuraMG** [comment]
yea.. oh no :( This happened on another project when we did our linting.. I can have a deeper look later today on this!

File: src/entities/_baseEntity.js [hunk 3]
```diff
@@ -1,37 +1,32 @@
...
-    return ({
-      type: this.type,
-      location: this.location,
-      coordinates: this.coordinates
-    })
-  };
-
-    this.getLocation = function(object, locArgs, canvasX, canvasY) { // eslint-disable-line
+  isMember() {
+    // TODO: What is this empty method for?
```
[the remark this answers is not in the document]
Comment (MathuraMG) [reply src/entities/_baseEntity.js]: @spiray  - I'm so sorry, I reviewed this a week ago, and did not confirm the review submission

File: src/entities/backgroundEntity.js [hunk 4]
```diff
@@ -1,16 +1,16 @@
-function BackgroundEntity(Interceptor, object, backgroundArgs, canvasX, canvasY) { // eslint-disable-line no-unused-vars
-  this.populate = function(Interceptor) {
-    if (backgroundArgs[0].name === `p5.Color`) {
-      backgroundArgs = backgroundArgs[0].levels;
+class BackgroundEntity {
+  constructor(Interceptor, backgroundArgs) {
+    this.backgroundArgs = [...backgroundArgs];
```
[the remark this answers is not in the document]
Comment (MathuraMG) [reply src/entities/backgroundEntity.js]: @spiray  - I'm so sorry, I reviewed this a week ago, and did not confirm the review submission

[the remark this answers is not in the document]
Comment (MathuraMG) [reply src/entities/backgroundEntity.js]: @spiray - regd the above issue, do you want to jump on a call and finish it up? I feel like thats always faster :)

File: src/entities/shapeEntity.js [hunk 5]
```diff
@@ -1,58 +1,35 @@
...
-    else {
-      return ({
-        type: this.type,
-        location: this.location,
-        coordinates: this.coordinates         
-      })
-    }
-  };
-
-  /* return length of lines */
```
Comment (MathuraMG) [inline src/entities/shapeEntity.js]: Looks like the special condition of lines was deleted during the merge

Comment (spiray) [reply src/entities/shapeEntity.js]: Fixed in next commit

**Comment by spiray** [comment]
@MathuraMG Can you merge this PR? I made the final changes and tested the examples and everything seems to be working well.

File: src/entities/shapeEntity.js [hunk 6]
```diff
@@ -1,58 +1,62 @@
...
-  this.length = 0;
-
-
-  this.populate = function(shapeObject, arguments, canvasX, canvasY) {
-    this.location = this.getLocation(shapeObject, arguments, canvasX, canvasY);
-    this.coordLoc = this.canvasLocator(shapeObject, arguments, canvasX, canvasY);
-    if(!shapeObject.name.localeCompare(`ellipse`) || !shapeObject.name.localeCompare(`rect`)  || !shapeObject.name.localeCompare(`triangle`) || !shapeObject.name.localeCompare(`quad`)) {
-      this.areaAbs = this.getObjectArea(shapeObject.name, arguments);
-      this.area = (this.getObjectArea(shapeObject.name, arguments) * 100 / (canvasX * canvasY)).toFixed(2) + `%`;
-    } else if(!shapeObject.name.localeCompare(`line`)) {
```
Comment (MathuraMG) [inline src/entities/shapeEntity.js]: @spiray - looks like this is still not merged in..

**Comment by MathuraMG** [comment]
@spiray left a comment..

File: src/entities/shapeEntity.js (hunk 6 above)
Comment (spiray) [reply src/entities/shapeEntity.js]: Fixed!

**Pull request merged**